forge-game/src/main/java/forge/game/ability/effects/SacrificeEffect.java

```diff
@@ -96,11 +96,15 @@ public void resolve(SpellAbility sa) {
         CardZoneTable zoneMovements = AbilityKey.addCardZoneTableParams(params, sa);
 
         if (valid.equals("Self") && game.getZoneOf(host) != null) {
-            if (game.getZoneOf(host).is(ZoneType.Battlefield)) {
-                if (!optional || activator.getController().confirmAction(sa, null,
-                        Localizer.getInstance().getMessage("lblDoYouWantSacrificeThis", host.getName()), null)) {
-                    if (game.getAction().sacrifice(host, sa, true, params) != null && remSacrificed) {
-                        host.addRemembered(host);
+            valid = sa.hasParam("Defined") ? sa.getParam("Defined") : valid;
+            List<Card> toSac = AbilityUtils.getDefinedCards(host, sa.getParam("Defined"), sa);
+            for (Card c : toSac) {
```

Comment:
traditionally (and somewhat weirdly) we've used SacrificeAll for Defined...
I guess you want to move away from it?

Reply:
Yeah, every time it just seems counterintuitive

I can bring it up in the discord to see if others mind

Reply:
I don't know if we should use DefinedCard there or not ...

In Theory, we could use both ways for `getTargetPlayers(sa)` & `getTargetCards(sa)` but that would mean a bigger rewrite

Reply:
So would it be better to:
 - Leave this "DefinedCard" for now and refactor when we're more sure what to do
 - Just use SacrificeAllEffect and save this whole discussion for a rainy day
 - Try to refactor SacrificeEffect in this PR (I'm good to try, but don't want to do it if we aren't ready yet)

Reply:
imo "Just use SacrificeAllEffect and save this whole discussion for a rainy day"

we could do it later